fix wording (#2373)
arbor-sim/arbor: doc/dev/mechanism_abi.rst

  .. c:member:: arb_size_type      partition_width

    granularity for this backed, e.g., SIMD lanes

  Interface methods:

  .. c:member:: arb_mechanism_method init_mechanism

    - called once during instantiation,
    - setup initial state, corresponds to NMODL's INITIAL block,
    - will receive an allocated and initialised pack object

  .. c:member:: arb_mechanism_method compute_currents

    - compute ionic currents and set them through pointers in `ion_state`, currents
      live in `current_density`

>     - called during each integration time step

Reviewer: ```suggestion
    - called during each integration time step, and
```

>       - at the start for reversal potential mechanisms, *before* current reset

Reviewer: ```suggestion
      - reversal potential mechanisms: at the beginning of the time step, *before* current reset
```

>       - after event deliver for anything else [a bit hard to read]

Reviewer: ```suggestion
      - after event delivery, if enabled, e.g. by defining ``NET_RECEIVE`` in NMODL
```

You can create mechanisms with both ``i_gpu`` and ``i_cpu`` returning ``null``,
but at least one of the interfaces must be provided or Arbor will refuse to load

> the catalogue of this mechanism.

Reviewer: ```suggestion
the catalogue containing this mechanism.
```
